homeassistant/components/sunweg/__init__.py

```diff
@@ -0,0 +1,20 @@
+"""The Sun WEG inverter sensor integration."""
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import PLATFORMS
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: config_entries.ConfigEntry
+) -> bool:
+    """Load the saved entities."""
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+    return True
```

Comment:
You should be creating the api here, and checking authentication then return configentrynotready or configentryauthfailed.


I've not seen someone not use a coordinator, so I don't know if that is allowed(could very well be, but i'm not sure). I
Then you should store the api/coordinator(if you need to make one) in hass.data[DOMAIN][entry.entry_id]

Reply:
Okay I think instead of a coordinator, you could use your  SunWEGData object.